source/crud/tests/unified/estimatedDocumentCount-comment.yml

```diff
@@ -0,0 +1,116 @@
+description: "estimatedDocumentCount-comment"
+
+schemaVersion: "1.0"
+
+createEntities:
+  - client:
+      id: &client0 client0
+      observeEvents: [ commandStartedEvent ]
+  - database:
+      id: &database0 database0
+      client: *client0
+      databaseName: &database0Name edc-comment-tests
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: &collection0Name coll0
+
+initialData:
+  - collectionName: *collection0Name
+    databaseName: *database0Name
+    documents:
+      - { _id: 1, x: 11 }
+      - { _id: 2, x: 22 }
+      - { _id: 3, x: 33 }
+
+tests:
+  - description: "estimatedDocumentCount with document comment - post 4.9"
+    runOnRequirements:
+      - minServerVersion: "4.9.0"
+    operations:
+      - name: estimatedDocumentCount
+        object: *collection0
+        arguments:
+          comment: &documentComment { key: "value"}
+        expectResult: 3
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                aggregate: *collection0Name
+                pipeline: &pipeline
+                  - $collStats: { count: {} }
+                  - $group: { _id: 1, n: { $sum: $count }}
+                comment: *documentComment
+              commandName: aggregate
+              databaseName: *database0Name
+
+  - description: "estimatedDocumentCount with document comment - pre 4.9"
+    runOnRequirements:
+      # https://jira.mongodb.org/browse/SERVER-63315
+      # Server supports count with comment of any type for comment starting from 4.4.14.
+      - minServerVersion: "4.4.14"
+        maxServerVersion: "4.8.99"
+    operations:
+      - name: estimatedDocumentCount
+        object: *collection0
+        arguments:
+          comment: *documentComment
+        expectResult: 3
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                count: *collection0Name
+                comment: *documentComment
+              commandName: count
+              databaseName: *database0Name
+
+  - description: "estimatedDocumentCount with string comment - pre 4.9"
```

Comment:
Just to confirm: did you intentionally omit a "string comment - post 4.9" test? I suppose it's redundant given "estimatedDocumentCount with document comment - post 4.9", since there's no reason `aggregate` wouldn't support both types (noting `document` is just a stand-in for _any_ BSON type).

Reply:
That is correct. I think it is safe to assume that if a test with a document comment pass, it means that this server accepts any BSON type as a comment.
We can add a test for every BSON type as a comment, but I think it would be too much.